blocktx/peer_handler.go

```diff
@@ -238,6 +239,21 @@ func (bs *PeerHandler) HandleTransaction(msg *wire.MsgTx, peer p2p.PeerI) error
 }
 
 func (bs *PeerHandler) HandleBlockAnnouncement(msg *wire.InvVect, peer p2p.PeerI) error {
+	primaryBlocktx, err := bs.store.PrimaryBlocktx(context.TODO())
```

Comment:
If this whole block is in a function, then it can be reused in both `HandleBlockAnnouncement` and `HandleBlock`
```
primaryBlocktx, err := bs.store.PrimaryBlocktx(context.TODO())
	if err != nil {
		return err
	}

	hostName, err := os.Hostname()
	if err != nil {
		return err
	}

	if primaryBlocktx != hostName {
		bs.logger.Infof("Not primary, skipping block processing")
		return nil
	}
```